File: examples/alexa-hello-world-java/src/test/java/com/amazon/ask/helloworld/handlers/AlexaApplicationTest.java
package com.amazon.ask.helloworld.handlers;

import com.amazon.ask.dispatcher.request.handler.HandlerInput;
import com.amazon.ask.model.IntentRequest;
import com.amazon.ask.model.RequestEnvelope;
import com.amazon.ask.model.Response;
import com.amazon.ask.model.ui.OutputSpeech;
import io.micronaut.test.annotation.MicronautTest;
import static org.junit.jupiter.api.Assertions.*;
import org.junit.jupiter.api.Test;
import javax.inject.Inject;
import java.util.Optional;

@MicronautTest
public class AlexaApplicationTest {

    @Inject
    AlexaApplication handler;

    @Test
    void testHelloWorldIntent() throws Exception {
        final HandlerInput.Builder builder = HandlerInput.builder();
        final RequestEnvelope.Builder envelopeBuilder = RequestEnvelope.builder();
        envelopeBuilder.withRequest(IntentRequest.builder().build());
        builder.withRequestEnvelope(envelopeBuilder.build());
        final Optional<Response> response = handler.greet(builder.build());
        assertTrue(response.isPresent());
        final OutputSpeech outputSpeech = response.get().getOutputSpeech();
        assertTrue(
                outputSpeech.toString().contains(
                        "Hello world"
                )
        );
    }
}

File: examples/alexa-hello-world-kotlin/src/test/kotlin/com/amazon/ask/helloworld/handlers/AlexaApplicationTest.kt
package com.amazon.ask.helloworld.handlers

import com.amazon.ask.dispatcher.request.handler.HandlerInput
import com.amazon.ask.model.IntentRequest
import com.amazon.ask.model.RequestEnvelope
import io.micronaut.test.annotation.MicronautTest
import org.junit.jupiter.api.Assertions.*
import org.junit.jupiter.api.Test
import javax.inject.Inject

@MicronautTest
class AlexaApplicationTest {

    @Inject
    lateinit var handler : AlexaApplication

    @Test
    fun testHelloWorldIntent()  {
        val builder = HandlerInput.builder()
        val envelopeBuilder = RequestEnvelope.builder()
        envelopeBuilder.withRequest(IntentRequest.builder().build())
        builder.withRequestEnvelope(envelopeBuilder.build())

        val response = handler.greet(builder.build())
        assertTrue(response.isPresent)

        val outputSpeech = response.get().outputSpeech
        assertTrue(
                outputSpeech.toString().contains(
                        "Hello world"
                )
        )
    }
}
